src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -592,96 +592,136 @@ namespace smt::noodler {
+
+        std::vector<LenNode> result_conjuncts;
+        for (const std::tuple<BasicTerm, BasicTerm, TransformationType>& transf : transformations) {
+            BasicTerm result = std::get<0>(transf);
+            BasicTerm argument = std::get<1>(transf);
+            TransformationType type = std::get<2>(transf);
+            switch (type)
+            {
+            case TransformationType::FROM_CODE:
+                STRACE("str-transform",
+                    tout << " procesing from_code with result " << result << " and argument " << argument << " which is handled by" << std::flush;
+                );
+                std::swap(result, argument);
+                // fall trough, we do nearly the same thing
+            case TransformationType::TO_CODE:
+            {
+                STRACE("str-transform",
+                    tout << " procesing to_code with result " << result << " and argument " << argument << std::endl;
+                );
+                /* Having result=to_code(argument) we need to take all var_1 ... var_n
+                 * substituting argument in solution. We need to have one of the |var_i| = 1
+                 * and all others |var_j| = 0; var_i will be then the char whose code point
+                 * we want to return. We keep this code point in int variable var_i!to_code,
+                 * where result = var_i!to_code (we do this, so that different transformations
+                 * are connected - they use the same !to_code variables).
+                 * If this does not happen (i.e. argument is not word of length one), the result
+                 * is equal to -1.
+                 * 
+                 * We also handle from_code here, we swapped arugment and result so we have
+                 * argument = from_code(result). We can do exactly the same, we compute the
+                 * code point from which we are constructing the char by checking possible
+                 * chars in solution. The only difference is that if result is not a valid
+                 * code point, then argument must be an empty string.
+                 */
+                Mata::Nfa::Nfa sigma_aut = solution.aut_ass.sigma_automaton();
+                std::vector<BasicTerm> substituted_vars = solution.get_substituted_vars(argument);
+
+                // Disjunction representing that result is equal to code point of one of the chars of some var_i
+                LenNode result_solution(LenFormulaType::OR, {});
+                for (const BasicTerm& var : substituted_vars) {
+                    // disjunction that will say that var!to_code is equal to code point of one of the chars in var
+                    LenNode to_code_disjunction = LenNode(LenFormulaType::OR, {});
+                    for (Mata::Symbol s : Mata::Strings::get_one_symbol_words(*solution.aut_ass.at(var))) { // iterate trough chars of var
```

Comment:
Yes, but it is a bit complicated. This function is already merged into `devel`, so after merging this, we should immediately update noodler so that it works with current version of mata.

Reply:
ok